pkg/client/jenkins.go

```diff
@@ -130,6 +131,10 @@ func (j JenkinsAPIConnectionSettings) Validate() error {
 		return errors.New("empty hostname is now allowed. Please provide hostname")
 	}
 
+	if (j.Portname == "" && j.Port > 0) || (j.Portname == "" && j.UseNodePort) {
```

Comment:
We should avoid to force a port name, we don't need it to let the operator or jenkins run, we just need it in the CRDs to be added in the helm chart and consumed by the operator while creating the jenkins pod

Reply:
@brokenpip3 I have made the required changes. Can you please review it? (sorry for the late reply I was busy preparing for GSoC'23)

Reply:
Like this that field is not consumed by the operator while creating the jenkins pod, you need to add some code in `pkg/configuration/base/resources/service.go` where also you need to add the default `http` as port name if is not defined in the crd :)

Reply:
Okay I will make the changes.